### PR TITLE
refactor npm install (again)

### PR DESCRIPTION
**Pull request by Narretz**

### CODE REVIEW

**Comment by petebacondarwin** [comment]
@IgorMinar  - can you take a quick look at this when you get a chance?

File: lib/grunt/utils.js [hunk 1]
```diff
@@ -6,11 +6,39 @@ var shell = require('shelljs');
...
+
+function shrinkwrapChanged() {
+  return shell.exec('diff ' + SHRINKWRAP_FILE + ' ' + SHRINKWRAP_CACHED_FILE, {silent: true}).output;
+}
+
+function cleanNodeModules() {
+  if (process.platform == "win32") {
+    rimraf.sync('node_modules');
+    console.log('cleaned node_modules using rimraf');
+  } else {
```
Comment (mzgol) [inline lib/grunt/utils.js]: Why this `else`? `rimraf` is cross-platform.

Comment (Narretz) [inline lib/grunt/utils.js]: I think rm -rf might still be more robust, so we should use it if possible. @IgorMinar should probably have a word here.

Comment (petebacondarwin) [inline lib/grunt/utils.js]: @IgorMinar wanted to use `rm -rf` where possible for performance reasons

**Comment by Narretz** [comment]
This has been confirmed to work on Windows in https://github.com/angular/angular.js/issues/11143 so I'm gonna merge it at the end of the week unless somebody has any objections.

**Comment by Narretz** [comment]
Although I'd like @IgorMinar to take a quick look. :)

**Comment by petebacondarwin** [comment]
Let's go with this @Narretz

**Comment by Narretz** [comment]
Aah, travis was actually using the install-dependencies.sh script. We need to fix this first.

**Comment by Narretz** [comment]
@petebacondarwin do you have an idea how we could handle this? I fear we have to duplicate the functionality, because we can't use the js file when travis is installing.

**Comment by petebacondarwin** [comment]
Hmm. That is a pain. And of course this doesn't work if `npm install` has not already been run as `rimraf` would not be available anyway.
Perhaps we just need an option not to use this at all on Windows?

**Comment by mzgol** [comment]
> I fear we have to duplicate the functionality, because we can't use the js file when travis is installing.

Why can't we? Node is already available, instead of:
```yaml
  - scripts/npm/install-dependencies.sh
```
we'd just use:
```yaml
  - node scripts/npm/install-dependencies.js
```
What am I not seeing here?

**Comment by petebacondarwin** [comment]
@mzgol the problem is that the new version of the `install-dependencies.xxx`, which happens to be a Grunt task now, relies upon things like:

```js
var rimraf = require('rimraf');
```

which of course will fail if we have not run `npm install` at least once already as that module will not yet exist locally.

Or am I missing something?

**Comment by mzgol** [comment]
@petebacondarwin Right... Could we just include `rimraf` manually in the repository and require it from this specific folder instead of `node_modules`? It's just 412 KB.

Or we could copy over the [rimraf.js file](https://github.com/isaacs/rimraf/blob/master/rimraf.js) and edit out `glob` which adds most of these 412 KB since we don't use glob patterns in our command (it would make it harder to upgrade the package later, though).

**Comment by petebacondarwin** [comment]
On non-Windows machines we can just use `rm -rf`. It is only on Windows machines that we need an alternative solution. So I think that we should try to do the switch at a higher level.

**Comment by petebacondarwin** [comment]
Closing in favour of #12792

**Pull request closed without merging**